test/fw/ptl/utils/plugins/ptl_test_data.py

```diff
@@ -149,160 +149,140 @@ def __save_home(self, test, status, err=None):
-                               sudo=True, level=logging.DEBUG2)
+
+            snap_out = ret['out'][0].strip(" ")
+            output_tar = snap_out.split(":")[1]
+
+        snap_out_dest = os.path.join(datadir, os.path.basename(snap_out))
+
+        snap_out_r = output_tar.strip(" ")
+        ret = self.du.run_copy(current_host, snap_out_r, snap_out_dest,
+                               level=logging.DEBUG2)
         if ret['rc'] != 0:
             _msg = 'Failed to copy generated diag from'
-            _msg += ' %s to %s' % (diag_out_r, diag_out_dest)
+            _msg += ' %s to %s' % (snap_out_r, snap_out_dest)
             f.write(_msg + '\n')
             self.logger.error(_msg)
             f.close()
             return
         else:
-            self.du.rm(svr_host, path=diag_out, sudo=True, force=True,
+            self.du.rm(server_host, path=snap_out, force=True,
                        level=logging.DEBUG2)
-        cores = []
-        dir_list = ['server_priv', 'sched_priv', 'mom_priv']
-        for d in dir_list:
-            path = os.path.join(svr.pbs_conf['PBS_HOME'], d)
-            files = self.du.listdir(hostname=svr_host, path=path, sudo=True,
-                                    level=logging.DEBUG2)
-            for _f in files:
-                if os.path.basename(_f).startswith('core'):
-                    cores.append(_f)
-        cores = list(set(cores))
-        if len(cores) > 0:
-            cmd = ['gunzip', diag_out_dest]
-            ret = self.du.run_cmd(current_host, cmd, sudo=True,
-                                  level=logging.DEBUG2)
-            if ret['rc'] != 0:
-                _msg = 'Failed unzip generated diag at %s:' % diag_out_dest
-                _msg += '\n\n' + '\n'.join(ret['err']) + '\n\n'
-                f.write(_msg + '\n')
-                self.logger.error(_msg)
-                f.close()
-                return
-            diag_out_dest = diag_out_dest.rstrip('.gz')
-            cmd = ['tar', '-xf', diag_out_dest, '-C', datadir]
-            ret = self.du.run_cmd(current_host, cmd, sudo=True,
-                                  level=logging.DEBUG2)
-            if ret['rc'] != 0:
-                _msg = 'Failed extract generated diag %s' % diag_out_dest
-                _msg += ' to %s:' % datadir
-                _msg += '\n\n' + '\n'.join(ret['err']) + '\n\n'
-                f.write(_msg + '\n')
-                self.logger.error(_msg)
-                f.close()
-                return
-            self.du.rm(hostname=current_host, path=diag_out_dest,
-                       force=True, sudo=True, level=logging.DEBUG2)
-            diag_out_dest = diag_out_dest.rstrip('.tar')
-            for c in cores:
-                cmd = [pbs_diag, '-g', c]
-                ret = self.du.run_cmd(svr_host, cmd, sudo=True,
-                                      level=logging.DEBUG2)
+        snap_out_dest = snap_out_dest.rstrip('.tgz')
```

Comment:
Why we are extracting snapshot tarball? previously in diag code, it was not extracting final diag tarball so I suggest we should keep that way... and let the caller of pbs_benchpress decide what to do with snapshot tarball.

NOTE: don't get confused with extracting at line no 220 in old code, it was there to extract backtrace of core file and remove large core file from diag tarball

Reply:
The reason behind untarig the snapshot tarball is if we capture logs on multinode setup then pbs_snapshot stores remote host logs inside localhost logs.
snapshot_datestamp_logs.tar.gz has below directories inside when we run on multinode setup having one remote host mom
- server_logs
- comm_logs
- etc
- remote_host_snapshot_log.tar.gz

In above directory structure, you can see remote_host tar ball. When TH2 extracts main snapshot tar file then it will have this remote host tar file and we cannot just open it and see the logs. That's why I am untarring the main tar file and extracting the tar files inside and renaming them.

Reply:
@shilpakodli I know this and that's why I told that let the caller of the pbs_benchpress handle that, why PTL has to take care of it? PTL will simply give one tarball which has all info and caller will do whatever it wants to do with that tarball. If we want PTL to give extracted tarball then why pbs_snapshot is storing data in a tarball at all? we should change that right? But we don't want that :smile:

Reply:
I agree with Hiren, PTL should just generate snapshot tarball, which might be used by external apps (like TH2) in whatever way they want. There's a lot of code here which untars the snapshot and the sub-snapshots inside it, I think you should remove all of it and instead add it to TH2.